apicurio-registry/pkg/controller/apicurioregistry/deployment_cf.go

```diff
@@ -0,0 +1,102 @@
+package apicurioregistry
+
+import (
+	"context"
+	ar "github.com/apicurio/apicurio-operators/apicurio-registry/pkg/apis/apicur/v1alpha1"
+	apps "k8s.io/api/apps/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var _ ControlFunction = &DeploymentCF{}
+
+type DeploymentCF struct {
+	ctx *Context
+}
+
+func NewDeploymentCF(ctx *Context) ControlFunction {
+
+	err := ctx.c.Watch(&source.Kind{Type: &apps.Deployment{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &ar.ApicurioRegistry{},
+	})
+
+	if err != nil {
+		panic("Error creating Deployment watch.")
+	}
+
+	return &DeploymentCF{ctx: ctx}
+}
+
+func (this *DeploymentCF) Describe() string {
+	return "Deployment Creation"
+}
+
+func (this *DeploymentCF) Sense(spec *ar.ApicurioRegistry, request reconcile.Request) error {
+	// Try to check if there is an existing deployment resource
+	deploymentName := this.ctx.configuration.GetConfig(CFG_STA_DEPLOYMENT_NAME)
+
+	deployments, err := this.ctx.kubecl.client.AppsV1().Deployments(this.ctx.configuration.GetSpecNamespace()).List(
```

Comment:
I do not get this. The operator expects to have multiple deployments per `ApicurioRegistry` instance?

If answer is not, you can request the deployment by specific name. The name can be some prefix and apicurioregistry name if multiple  `ApicurioRegistry` instances are allowed in a single namespace.